src/toolkits/drawing_classifier/drawing_classifier.hpp

```diff
@@ -0,0 +1,250 @@
+             variant_type validation_data,
+             std::map<std::string, flexible_type> opts);
+  gl_sarray predict(gl_sframe data, std::string output_type);
+  gl_sframe predict_topk(gl_sframe data, std::string output_type,
+    size_t k, size_t batch_size);
+  variant_map_type evaluate(gl_sframe data, std::string metric,
+    size_t batch_size);
+  std::shared_ptr<coreml::MLModelWrapper> export_to_coreml(
+      std::string filename);
+  void load_pretrained_model(variant_map_type model_data, size_t version);
+
+  BEGIN_CLASS_MEMBER_REGISTRATION("drawing_classifier")
+
+  IMPORT_BASE_CLASS_REGISTRATION(ml_model_base);
+
+  REGISTER_CLASS_MEMBER_FUNCTION(drawing_classifier::train, "data",
+                                 "target_column_name", "feature_column_name",
+                                 "validation_data", "options");
+  register_defaults("train",
+                    {{"validation_data", to_variant(std::string("auto"))},
+                     {"options",
+                      to_variant(std::map<std::string, flexible_type>())}});
+  REGISTER_CLASS_MEMBER_DOCSTRING(
+      drawing_classifier::train,
+      "----------\n"
+      "data : SFrame\n"
+      "    Input data, which consists of columns named by the\n"
+      "    feature_column_name and target_column_name parameters, used for\n"
+      "    training the Drawing Classifier."
+      "target_column_name : string\n"
+      "    Name of the column containing the target variable. The values in "
+      "    this column must be of string type.\n"
+      "feature_column_name : string\n"
+      "    Name of the column containing the input drawings.\n"
+      "    The feature column can contain either bitmap-based drawings or\n"
+      "    stroke-based drawings. Bitmap-based drawing input can be a\n"
+      "    grayscale tc.Image of any size.\n"
+      "\n"
+      "    Stroke-based drawing input must be in the following format:\n"
+      "    Every drawing must be represented by a list of strokes, where each\n"
+      "    stroke must be a list of points in the order in which they were\n"
+      "    drawn on the canvas.\n"
+      "\n"
+      "    Each point must be a dictionary with two keys,\n"
+      "    \"x\" and \"y\", and their\n"
+      "    respective values must be numerical, i.e. either integer or float.\n"
+      "validatation_data : SFrame or string\n"
+      "    A dataset for monitoring the model's generalization performance to\n"
+      "    prevent the model from overfitting to the training data.\n"
+      "\n"
+      "    For each row of the progress table, accuracy is measured over the\n"
+      "    provided training dataset and the `validation_data`. The format of\n"
+      "    this SFrame must be the same as the training set.\n"
+      "\n"
+      "    When set to 'auto', a validation set is automatically sampled from "
+      "the\n"
+      "    training data (if the training data has > 100 sessions).\n"
+      "warm_start : string\n"
```

Comment:
Look at how the OD class works in this case. Chances are for now we just want to pass a path to the pre-trained model on disk. This we can always add later anyway, we can focus for now on getting cold-start working

Reply:
Gonna take it out, since we should think about this properly at a later point.